### PR TITLE
[[ Merge 6.7.7 rc 1 ]]

### PR DESCRIPTION
**Pull request by livecodesebastien**
Refactored:
    MCGroup::setprop addition of P_INVISIBLE and P_VISIBLE properties

### CODE REVIEW

File: engine/src/group.cpp [hunk 1]
```diff
@@ -3652,3 +3690,23 @@ void MCGroup::scheduledelete(bool p_is_child)
...
+// PM-2015-07-02: [[ Bug 13262 ]] Make sure we attach/detach the player when
+//  showing/hiding a group that has a player
+void MCGroup::SetVisibility(MCExecContext &ctxt, uinteger_t part, bool flag, bool visible)
+{
+#ifdef PLATFORM_PLAYER
+    for(MCPlayer *t_player = MCplayers; t_player != nil; t_player = t_player -> getnextplayer())
+    {
+        if (t_player -> getparent() == this)
+        {
+            if (visible)
```
Comment (livecodeali) [inline engine/src/group.cpp]: This should be 
`if (visible && flag) || (!visible && !flag)`
or perhaps 
`if (visible == flag)`
although it's probably less clear what's going on in the latter example.

Comment (peter-b) [inline engine/src/group.cpp]: I'm confused -- in this context, `visible` seems to be a `bool` function parameter, rather than a flag word...?

Comment (peter-b) [inline engine/src/group.cpp]: Derp. I can't read. :dizzy_face:

Comment (livecodeali) [inline engine/src/group.cpp]: visible is the (badly named) parameter determining which property is being set - true -> P_VISIBLE, false -> P_INVISIBLE

The flag param is what the property is being set to.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok af33e81

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok af33e81a699a4d7f9efc5a2feff62c12944bfcb9

**Comment by livecode-vulcan** [comment]
😎 test success af33e81a699a4d7f9efc5a2feff62c12944bfcb9
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/435)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/495)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/381)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/446)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/381)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/372)

**Pull request merged**